File: spring_java_orm_sample/src/main/java/com/example/demo/doma/entity/BookWithPublisherName.java
package com.example.demo.doma.entity;

import lombok.Getter;
import lombok.Setter;
import org.seasar.doma.Association;
import org.seasar.doma.Column;
import org.seasar.doma.Entity;
import org.seasar.doma.Id;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
public class BookWithPublisherName {
    @Id
    @Column(name = "id")
    Long id;

    @Column(name = "title")
    String title;

    @Column(name = "author")
    String author;

    @Column(name = "release_date")
    LocalDate releaseDate;

    @Column(name = "publisher_id")
    Long publisherId;

    @Column(name = "publisher_name")
    String publisherName;

    @Column(name = "genre_id")
    Long genreId;

    @Column(name = "genre_name")
    String genreName;

    @Column(name = "update_at")
    LocalDateTime updateAt;

    @Column(name = "version")
    Long version;

    @Association
    List<BookStockWithStoreName> bookStockList = new ArrayList<>();
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/doma/entity/BookWithPublisherName.kt
package com.example.demo.doma.entity

import org.seasar.doma.Association
import org.seasar.doma.Column
import org.seasar.doma.Entity
import org.seasar.doma.Id
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
class BookWithPublisherName {
    @Id
    @Column(name = "id")
    var id: Long? = null

    @Column(name = "title")
    var title: String? = null

    @Column(name = "author")
    var author: String? = null

    @Column(name = "release_date")
    var releaseDate: LocalDate? = null

    @Column(name = "publisher_id")
    var publisherId: Long? = null

    @Column(name = "publisher_name")
    var publisherName: String? = null

    @Column(name = "genre_id")
    var genreId: Long? = null

    @Column(name = "genre_name")
    var genreName: String? = null

    @Column(name = "isbn")
    var isbn: String? = null

    @Column(name = "sales_unit_price")
    var salesUnitPrice: Int? = null

    @Column(name = "update_at")
    var updateAt: LocalDateTime? = null

    @Column(name = "version")
    var version: Long? = null

    @Association
    var bookStockList: MutableList<BookStockWithStoreName?>? = ArrayList<BookStockWithStoreName?>()
}
